### PR TITLE
Update WASM 0.2.0, Refactor and Zmenu fix

### PR DESCRIPTION
**Pull request by msmanoj**
### Description
- Updated WASM to 0.2.0
- Separated the functions within the main class file into separate files

I have also pulled in all the changes from https://github.com/Ensembl/ensembl-genome-browser/pull/2

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -73,4 +73,27 @@ These are the actions that are dispatched from the GB to the external applicatio
...
 
 ## Links
-- Peregrine Repo: https://github.com/Ensembl/ensembl-dauphin-style-compiler
\ No newline at end of file
+- Peregrine Repo: https://github.com/Ensembl/ensembl-dauphin-style-compiler
+
+
+## For Ensembl developers: publishing npm package to the Gitlab registry
+
+Npm is set in this repository to use the gitlab repository https://gitlab.ebi.ac.uk/ensembl-web/package-registry an an npm package registry. See the `publishConfig` field in `package.json` and the `.npmrc` file for the configuration.
```
Comment (azangru) [inline README.md]: `an an` -> `as an` (my bad)

**Comment by azangru** [comment]
Do we even want to keep the `dist` folder after publishing the package to gitlab registry?

**Comment by msmanoj** [comment]
> Do we even want to keep the `dist` folder after publishing the package to gitlab registry?

I don't think we need it. Can we add a script to delete it after publishing?

Something like: `"postpack": "rm -rf ./dist"` ?

**Comment by azangru** [comment]
> I don't think we need it. Can we add a script to delete it after publishing?
> Something like: "postpack": "rm -rf ./dist" ?

Hmm, I would personally add the dist folder to the  `.gitignore` file (after deleting it first, and committing the removal of this directory to git). We already have the `.npmignore` folder, so npm isn't going to exclude paths based on gitignore, which it would have done otherwise (see [comment on SO](https://stackoverflow.com/a/24942436/3925302)).

I don't know if this is better or worse than explicitly deleting the `dist` folder everytime in `postpack`; but I believe adding unwanted folders to gitignore is a standard solution for this.

File: README.md (hunk 1 above)
Comment (msmanoj) [reply README.md]: That's done

**Comment by msmanoj** [comment]
@azangru  - deleted dist and added it to `.gitignore`

**Review by azangru — APPROVED** [review]

**Pull request merged**